tensorflow/core/common_runtime/direct_session.cc

```diff
@@ -138,9 +138,10 @@ Status NewThreadPoolFromThreadPoolOptions(
   return OkStatus();
 }
 
-thread::ThreadPool* GlobalThreadPool(const SessionOptions& options) {
+thread::ThreadPool* GlobalThreadPool(const SessionOptions& options,
+                                     int32_t expected) {
   static thread::ThreadPool* const thread_pool =
-      NewThreadPoolFromSessionOptions(options);
+      NewThreadPoolFromSessionOptions(options, expected);
```

Comment:
Call this `num_threads` instead of `expected`, and document (via comment on the function) that  if it is <=0 then the number of threads will be parsed from the options.  The term  "expected" usually has implications that if the value is not what is expected then an error should be thrown.  That's not what we're doing here - instead, we're overriding the session options.